### DISCUSSION TITLE
converting a color in another (failure)

### DISCUSSION THREAD

**Discussion by CA10C3SAR** [ImageMagick/ImageMagick#2176] [Help] [answered]
![name0](https://user-images.githubusercontent.com/6724006/85186662-c20dff80-b270-11ea-80a6-bf8084f29cb5.png)
I'm trying to convert this image above, color c1c1c1 to 00c000, what I have tried:
`convert name0.png -fill "#00c000" -opaque "#c1c1c1" result.png`

But after running the code, the color remains the same, it's not changing, the result I would expect:
![teste](https://user-images.githubusercontent.com/6724006/85186731-14e7b700-b271-11ea-9889-8580706b254d.png)

**Comment by snibgo** [comment]
"identify -verbose" says it doesn't contain the colour #c1c1c1.

**Reply by fmw42** [reply]
Your colors do not perfectly match what is in your image. So you need to provide a small fuzz value to your command.

`convert mountain_troll.png -fuzz 10% -fill "#00c000" -opaque "#c1c1c1" result.png
`

![result](https://user-images.githubusercontent.com/13458218/85187130-3f764780-b252-11ea-8df7-c5135830d560.png)

**Reply by CA10C3SAR** [reply]
i got the image black now using -fuzz 10%:
![result](https://user-images.githubusercontent.com/6724006/85187196-49a93d80-b274-11ea-9df4-a4e71b27326b.png)
"Your colors do not perfectly match what is in your image"
I got the colors using photoshop:
![Photoshop_2020-06-19_21-32-40](https://user-images.githubusercontent.com/6724006/85187219-7198a100-b274-11ea-95e7-92d3f05e4c6d.png)
What colors didn't match?

**Reply by fmw42** [reply]
Your photoshop says 193, but other tools say 192.  192 corresponds to #C0C0C0.

identify -verbose mountain_troll.png

mountain_troll.png
Image:
  Filename: mountain_troll.png
  Format: PNG (Portable Network Graphics)
  Mime type: image/png
  Class: DirectClass
  Geometry: 91x9+0+0
  Resolution: 37.79x37.79
  Print size: 2.40804x0.238158
  Units: PixelsPerCentimeter
  Colorspace: Gray
  Type: GrayscaleAlpha
  Base type: Undefined
  Endianness: Undefined
  Depth: 8-bit
  Channel depth:
    gray: 8-bit
    alpha: 1-bit
  Channel statistics:
    Pixels: 819
    Gray:
      min: 192  (0.752941)
      max: 255 (1)
      mean: 232.692 (0.912519)
      standard deviation: 30.1473 (0.118225)
      kurtosis: -1.63101
      skewness: -0.609081
      entropy: 0.937668
    Alpha:
      min: 0  (0)
      max: 255 (1)
      mean: 90.293 (0.35409)
      standard deviation: 122.025 (0.478529)
      kurtosis: -1.63101
      skewness: 0.609081
      entropy: 0.937668
  Alpha: graya(255,0)   #FFFFFF00
  Colors: 2
  Histogram:
    **290: (192,192,192,255) #C0C0C0FF graya(192,1)  <-----**
    529: (255,255,255,0) #FFFFFF00 graya(255,0)

**Reply by CA10C3SAR** [reply]
i have tried using #c0c0c0ff and i'm still getting a black image as a result instead of green.
![result](https://user-images.githubusercontent.com/6724006/85188303-fb983800-b27b-11ea-8013-efc7e7c88b29.png)
I also run -verbose on expected image to check if the green color was correct, and its
https://youtu.be/tlL9f0QPgGw

**Reply by fmw42** [reply]
I cannot see anything in that video. Please post text from identify -verbose.

This works fine for me.

`convert mountain_troll.png -fill "#00c000" -opaque "#c0c0c0" result.png`

![result](https://user-images.githubusercontent.com/13458218/85188966-7ef76080-b25f-11ea-8cc5-f3faafc1ca4e.png)

However, I note that when I measure the colors in my Photoshop CC, it says:

`cdcdcd`

So I think different tools and/or different OS measure/convert to hex colors slightly differently or there is some precision issue.

**Comment by snibgo** [comment, marked as answer]
The input is grayscale. You need to promote it to 3 channels: "-colorspace sRGB" after reading it.
